lib/rubocop/cop/factory_girl/dynamic_attribute_defined_statically.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module FactoryGirl
+      # Prefer to declare dynamic attribute values in a block.
+      #
+      # @example
+      #   # bad
+      #   kind [:active, :rejected].sample
+      #
+      #   # good
+      #   kind { [:active, :rejected].sample }
+      #
+      #   # bad
+      #   closed_at 1.day.from_now
+      #
+      #   # good
+      #   closed_at { 1.day.from_now }
+      #
+      #   # good
+      #   kind :static
+      #
+      #   # good
+      #   comments_count 0
+      #
+      #   # good
+      #   type User::MAGIC
+      class DynamicAttributeDefinedStatically < Cop
+        MSG = 'Use a block to set a dynamic value to an attribute.'.freeze
+
+        def_node_matcher :dynamic_defined_statically?, <<-PATTERN
+          (send nil _method (send ... {:ago :current :from_now :now :sample}))
+        PATTERN
+
+        def on_send(node)
+          return if already_defined_in_a_block?(node)
+          return unless dynamic_defined_statically?(node)
+          add_offense(node, :expression)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            expression = node.descendants.first.location.expression
+            corrector.insert_before(expression, '{ ')
+            corrector.insert_after(expression, ' }')
+          end
+        end
+
+        private
+
+        def already_defined_in_a_block?(node)
+          node.sibling_index.zero? && node.parent && node.parent.block_type?
```

Comment:
I observed that the value is the first sibling when it's a block. I'll recheck if I can improve it.